content/concepts/fundamentals/connections.md

```diff
@@ -0,0 +1,203 @@
+efficient than opening a separate connection for each stream. This can be especially helpful
+in applications that simultaneously maintain many connections or streams, such as a peer-to-peer
+file-sharing application. Multiplexing also allows for better utilization of network resources
+and can improve overall performance. For example, if one stream is idle while another stream
+is receiving a burst of data, the idle stream will not block the active stream from using the
+connection.
+
+### Protocol support
+
+libp2p supports a variety of protocols for security and stream multiplexing, including:
+
+- Security protocols:
+  - noise: Provides encryption, authentication, and forward secrecy for libp2p connections.
+  - tls: Provides encryption, authentication, and forward secrecy for libp2p connections.
+
+Learn more about secure connections in the
+[secure communications section](../secure-comm/overview.md).
+
+- Stream multiplexing protocols:
+  - yamux: Provides multiplexing of streams over a single connection.
+  - mplex: Provides multiplexing of streams over a single connection.
+
+Learn more about multiplexing in the
+[stream multiplexing section](../multiplex/overview.md).
+
+New protocols can be added to libp2p as needed. When adding a new protocol, it is
+important to consider compatibility with existing protocols, performance, and security.
+
+### Upgrading connections
+
+libp2p is designed to support a variety of transport protocols, including those that do
+not natively support the core libp2p capabilities of security and stream multiplexing.
+The process of layering capabilities onto "raw" transport connections is known as
+"upgrading" the connection.
+
+The listener's multiaddr determines the security protocol to be used on a connection
+(see [addressing specification]). The multiplexing protocol is determined using protocol
+negotiation, with the multistream-select protocol used for this negotiation process
+(as described in the [Protocol Negotiation section](#protocol-negotiation)). When raw
+connections need both security and multiplexing, security is always established first,
+and the negotiation for stream multiplexing takes place over the encrypted channel.
+
+Here is an example of the connection upgrade process:
+
+- The dialing peer sends a request to initiate a connection to the listening
+  peer over the underlying transport protocol (e.g., TCP).
+- The listening peer accepts the incoming connection request and sends the security
+  protocol ID (e.g., Noise) in its multiaddr to indicate the security protocol to use.
+- The dialing peer responds with the security handshake message to initiate the Noise
+  protocol.
+- If the security handshake is successful, the peers exchange the multistream protocol
+  ID to establish that they will use multistream-select to negotiate protocols for the
+  connection upgrade.
+- The peers negotiate which stream multiplexer to use by proposing and responding with
```

Comment:
In some cases, the peer may include the stream muxer in the security handshake to save this roundtrip https://github.com/libp2p/specs/blob/master/connections/inlined-muxer-negotiation.md